test/e2e/config/vsphere.yaml

```
@@ -34,29 +34,29 @@ providers:
   - name: cluster-api
     type: CoreProvider
     versions:
-      - name: v1.7.0
+      - name: "{go://sigs.k8s.io/cluster-api@latest-v1.7}" # latest published release in the v1beta1 series
```

This here changes to `v1.7.0-rc.0` (or `rc.1`), which is why the init falls back to v1.6.3 for initialisation.

---

```suggestion
      - name: "v1.7.0" # latest published release in the v1beta1 series
```

Keeping these name fields to v1.7.0 as long as there are pre-releases should help. But we should find a better way IMHO 🤔

---

Thanks @chrischdi for the help! Yes, it should be pulling v1.7.0-rc.x with this PR, but I didn't quite get why it falls back to v1.6.3 and what do you mean about pre-releases?

---

It uses the latest non-pre-release version in the configuration.

Faking it to v1.7.0 does the trick here, although not being optimal.

---

Got it, thanks! So for this PR, I'll leave it as hard-coded.